internal/services/oracle/oracle_autonomous_database_clone_from_backup_resource.go

```diff
@@ -0,0 +1,411 @@
+				ComputeModel:                   pointer.To(autonomousdatabases.ComputeModel(model.ComputeModel)),
+				CustomerContacts:               pointer.To(expandCloneCustomerContacts(model.CustomerContacts)),
+				DataStorageSizeInTbs:           pointer.To(model.DataStorageSizeInTb),
+				DbWorkload:                     pointer.To(autonomousdatabases.WorkloadType(model.DatabaseWorkload)),
+				DbVersion:                      pointer.To(model.DatabaseVersion),
+				DisplayName:                    pointer.To(model.DisplayName),
+				IsAutoScalingEnabled:           pointer.To(model.AutoScalingEnabled),
+				IsAutoScalingForStorageEnabled: pointer.To(model.AutoScalingForStorageEnabled),
+				IsMtlsConnectionRequired:       pointer.To(model.MtlsConnectionRequired),
+				LicenseModel:                   pointer.To(autonomousdatabases.LicenseModel(model.LicenseModel)),
+				NcharacterSet:                  pointer.To(model.NationalCharacterSet),
+				WhitelistedIPs:                 pointer.To(model.AllowedIps),
+			}
+			properties := param.Properties.(*autonomousdatabases.AutonomousDatabaseFromBackupTimestampProperties)
+			if model.BackupTimestamp != "" {
+				properties.Timestamp = pointer.To(model.BackupTimestamp)
+			}
+			if model.SubnetId != "" {
+				properties.SubnetId = pointer.To(model.SubnetId)
+			}
+
+			if model.VnetId != "" {
+				properties.VnetId = pointer.To(model.VnetId)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCloneFromBackupResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+			id, err := autonomousdatabases.ParseAutonomousDatabaseID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			var state AutonomousDatabaseCloneFromBackupResourceModel
+
+			if v, ok := metadata.ResourceData.GetOk("use_latest_available_backup_time_stamp"); ok {
+				state.UseLatestAvailableBackupTimestamp = v.(bool)
+			}
+			if v, ok := metadata.ResourceData.GetOk("backup_timestamp"); ok {
+				state.BackupTimestamp = v.(string)
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Tags = pointer.From(model.Tags)
+				state.Name = id.AutonomousDatabaseName
+				state.ResourceGroupName = id.ResourceGroupName
+
+				props, ok := model.Properties.(autonomousdatabases.AutonomousDatabaseFromBackupTimestampProperties)
+				if !ok {
+					return fmt.Errorf("%s was not of type `CloneFromBackupTimestamp`", id)
+				}
```

Comment:
according to the `Read` example in the [guidance](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md#step-3-scaffold-an-emptynew-resource), please change it as below:

if props := model.Properties; props != nil {
...
}